File: latex/main paper/python code/RRMC_IVP.py
from random import random as U
def Y_in(t, tn, yn, h):
    S = tn + U()*(t-tn)  # \sim Uniform(T,t)
    return yn + h*Y_in(S, tn, yn, h) if U() < (t-tn)/h else yn
def Y_out(tn, h): # h is out step size
    TT = tn-h if tn-h > 0 else 0  
    return Y_in(tn, TT, Y_out(TT, h), h) if tn > 0 else 1

File: latex/main paper/julia_code/RRMC_IVP.jl
function Y_in(t, tn, yn, h)
    S = tn + rand() * (t - tn)  # \sim Uniform(T,t)
    return rand() < (t - tn) / h ? yn + h * Y_in(S, tn, yn, h) : yn
end
function Y_out(tn, h) # h is out step size
    TT = tn > h ? tn - h : 0
    return tn > 0 ? Y_in(tn, TT, Y_out(TT, h), h) : 1
end